src/compressFile.ts

```diff
@@ -0,0 +1,45 @@
+import * as fs from 'fs';
+import * as zlib from 'zlib';
+
+function compressFile(filename: string): Promise<void> {
+    const tempFilename = `${filename}.temp`;
+
+    fs.renameSync(filename, tempFilename);
+
+    const deleteFile = (file: string): void => {
+        try {
+            fs.unlinkSync(file);
+        } catch (_err) {
+            /* istanbul ignore next */
+        }
+    };
+
+    try {
+        return new Promise((resolve, reject) => {
```

Comment:
this will not work as expected.

An example of why it won't work
```ts
function f() {
  try {
    return new Promise(() => {
      throw new Error('broke');
    })
  } catch (e) {
    console.error('caught it', e);
  }
}

const x = f();
// x === Promise.reject('broke')
```

The better way to write this code is:
```ts
        const read = fs.createReadStream(tempFilename);
        const zip = zlib.createGzip();
        const write = fs.createWriteStream(filename);
        read.pipe(zip).pipe(write);
        await new Promise((resolve, reject) => {
            write.on(
                'error',
                /* istanbul ignore next */ err => {
                    write.end();
                    reject(err);
                },
            ); // close the write stream and propagate the error
            write.on('finish', () => {
                resolve();
            });
        });
```

Reply:
yeah, sure; honestly, I don't know how the stream creation and wiring stuff ended up within the `promise` code... should be fixed now...